### DISCUSSION TITLE
SSDT Custom Logger to exclude certain warnings/errors? (not just by code)

### DISCUSSION THREAD

**Discussion by chadbaldwin** [dotnet/msbuild#7859] [General] [answered]
### TL;DR

Is there a way to write a custom logger to suppress certain warnings and prevent them from showing up in Visual Studio? I want to suppress them by more than _just_ the code, because that is not granular enough for my current needs. I need like regex filtering or something.

----

### Long version

I have searched the internet far and wide for days only to come up empty handed. I've spent so many hours googling about this, I don't know what else to do at this point.

I am dealing with a MASSIVE SSDT project. Unfortunately, this project uses a lot of temp tables that are created in one proc and then "passed" down to a child proc. SSDT does not like this because the table is not created within that proc so it sees that table as an unresolved reference. The lowest granularity you get for filtering warning codes is at the file level, which is not helpful here.

I have thousands of these warnings (SQL71502). Unfortunately, the code used for these, is the same code used for all unresolved reference issues. So if I ignore this at the project level, I'm now ignoring many legitimate issues.

My hope...is to find some way to get MSBuild or Visual Studio to hide certain warnings based on some sort of glob/regex criteria.

Is this possible? Or will I be wasting my time if I try to go further down this road?

I have already written a PowerShell script that will parse MSBuild messages, and then filter out all the ones I don't care about...so I'm at least able to work through the issues that way. But unfortunately, it's not as nice as using the built in "live" view of issues in Visual Studio.

**Comment by rainersigwald** [comment, marked as answer]
Unfortunately, no: loggers are designed to not be able to interfere with each other, so while you can (and people do) write a logger that has a custom ignore-warning policy, it wouldn't affect the warnings themselves or what gets displayed in VS (which goes through the VS logger).

**Reply by chadbaldwin** [reply]
Well that's unfortunate. Do you happen to know if there is anything out there within the MSBuild world that may be a solution for what I'm trying to do?

I just stumbled upon this regarding custom regex parsing:
https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-diagnostic-format-for-tasks

But that doesn't sound useful. That sounds like it's just for custom parsing of a message, not for any sort of filtering.

I think I'm in over my head on this one unless I dedicate a large amount of time on a catching up on MSBuild internals.

**Reply by rainersigwald** [reply]
You could use that functionality if you have access to the ToolTask/Exec that is running the tool, but I don't know offhand if SSDT uses that for this level. If you can modify it, you could modify it to the default regex + `NOT my-special-condition`, but that's likely to be messy.

This just isn't something where the MSBuild engine offers extensibility for you, sorry.

**Reply by chadbaldwin** [reply]
Okay, I appreciate your insight and your time. Thanks!
